kernel/malloc.c

```diff
@@ -230,38 +240,38 @@ static void * f_sbrk(int flags, int incr)
         heap_stack = &_stack - 4096;
     }
 
-    if (flags & MEM_USER) {
+    if ((flags & MEM_USER) == MEM_USER) {
         if (!heap_end_user)
             return (void *)(0 - 1);
         /* Do not over-commit */
-        if ((heap_end_user + incr) > heap_limit_user)
+        if ((heap_end_user + incr) >= heap_limit_user)
             return (void *)(0 - 1);
         prev_heap_end = heap_end_user;
         heap_end_user += incr;
         memset(prev_heap_end, 0, incr);
 #ifdef CONFIG_SRAM_EXTRA
-    } else if (flags & MEM_EXTRA) {
+    } else if ((flags & MEM_EXTRA) == MEM_EXTRA) {
```

Comment:
This will never be entered, unless we change the value for `MEM_EXTRA` not to match `MEM_USER`